_inc/lib/core-api/class-wpcom-rest-field-controller.php

```diff
@@ -0,0 +1,326 @@
+	}
+
+	/**
+	 * Returns the schema's default value
+	 *
+	 * If there is no default, returns the type's falsey value.
+	 *
+	 * @param array $schema
+	 * @return mixed
+	 */
+	final public function get_default_value( $schema ) {
+		if ( isset( $schema['default'] ) ) {
+			return $schema['default'];
+		}
+
+		// If you have something more complicated, use $schema['default'];
+		switch ( isset( $schema['type'] ) ? $schema['type'] : 'null' ) {
+			case 'string' :
+				return '';
+			case 'integer' :
+			case 'number' :
+				return 0;
+			case 'object' :
+				return (object) array();
+			case 'array' :
+				return array();
+			case 'boolean' :
+				return false;
+			case 'null' :
+			default :
+				return null;
+		}
+	}
+
+	/**
+	 * The field's wrapped getter. Does permission checks and output preparation.
+	 *
+	 * This cannot be extended: implement `->get()` instead.
+	 *
+	 * @param mixed $object_data Probably an array. Whatever the endpoint returns.
+	 * @param string $field_name Should always match `->field_name`
+	 * @param WP_REST_Request $request
+	 * @param $object_type Should always match `->object_type`
+	 * @return mixed
+	 */
+	final public function get_for_response( $object_data, $field_name, $request, $object_type ) {
+		$permission_check = $this->get_permission_check( $object_data, $request );
+
+		if ( ! $permission_check || is_wp_error( $permission_check ) ) {
+			return $this->get_default_value( $this->get_schema() );
+		}
+
+		$value = $this->get( $object_data, $request );
+
+		return $this->prepare_for_response( $value, $request );
+	}
+
+	/**
+	 * The field's wrapped setter. Does permission checks.
+	 *
+	 * This cannot be extended: implement `->update()` instead.
+	 *
+	 * @param mixed $value The new value for the field.
+	 * @param mixed $object_data Probably a WordPress object (e.g., WP_Post)
+	 * @param string $field_name Should always match `->field_name`
+	 * @param WP_REST_Request $request
+	 * @param $object_type Should always match `->object_type`
+	 * @return void|WP_Error
+	 */
+	final public function update_from_request( $value, $object_data, $field_name, $request, $object_type ) {
+		$permission_check = $this->update_permission_check( $value, $object_data, $request );
+
+		if ( ! $permission_check ) {
+			/* translators: %s: get_permission_check() */
+			_doing_it_wrong( 'WPCOM_REST_API_V2_Field_Controller::update_permission_check', sprintf( __( "Method '%s' must return either true or WP_Error." ), 'update_permission_check' ), 'Jetpack 6.8' );
+			return;
+		}
+
+		if ( is_wp_error( $permission_check ) ) {
+			return $permission_check;
+		}
+
+		$updated = $this->update( $value, $object_data, $request );
+
+		if ( is_wp_error( $updated ) ) {
+			return $updated;
+		}
+	}
+
+	/**
+	 * Permission Check for the field's getter. Must be implemented in the inheriting class.
+	 *
+	 * @param mixed $object_data Whatever the endpoint would returnn for its response.
+	 * @param WP_REST_Request
```

Comment:
Should we specify the param name (`$request`) here? (Same for the next couple of methods below.)